src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs

```diff
@@ -647,15 +647,7 @@ private void LogRenameSession(RenameLogMessage.UserActionOutcome outcome, bool p
         }
     }
 
-    public void Cancel()
-    {
-        _threadingContext.ThrowIfNotOnUIThread();
-
-        // This wait is safe.  We are not passing the async callback to DismissUIAndRollbackEditsAndEndRenameSessionAsync.
-        // So everything in that method will happen synchronously.
-        DismissUIAndRollbackEditsAndEndRenameSessionAsync(
```

Comment:
Key change in the PR.
I feel we don't need `_threadingContext.ThrowIfNotOnUIThread();` because in `DismissUIAndRollbackEditsAndEndRenameSessionAsync` it immediately switches to main thread.